src/runners/search_index/search_index_runner.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ApiResponse } from '@opensearch-project/opensearch';
+import { ResponseError } from '@opensearch-project/opensearch/lib/errors';
+import { ApiProvider } from 'promptfoo';
+import { LevenshteinMatcher } from '../../matchers/levenshtein';
+import { PythonMatcher } from '../../matchers/python';
+import { openSearchClient } from '../../providers/clients/opensearch';
+import { OpenSearchProviderResponse } from '../../providers/types';
+import { OpenSearchTestIndices } from '../../utils/indices';
+import { TestResult, TestRunner, TestSpec } from '../test_runner';
+
+interface DSLSpec extends TestSpec {
+  question: string;
+  gold_query: string;
+  index: string;
+}
+
+interface DSLResponseItem {
+  _id: string;
+  _score: string;
+  _index: string;
+  _source: string;
+}
+interface DSLResponse {
+  hits: {
+    hits: Array<DSLResponseItem>;
+  };
+}
+
+class GoldQueryError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = 'GoldQueryError';
+  }
+}
+
+export class SearchIndexRunner extends TestRunner<DSLSpec, ApiProvider> {
+  levenshtein = new LevenshteinMatcher();
+  query_eval = new PythonMatcher('os_query_eval/eval.py');
+
+  protected async beforeAll(clusterStateId: string): Promise<void> {
+    await OpenSearchTestIndices.init();
+    await OpenSearchTestIndices.create(clusterStateId, { ignoreExisting: true });
+  }
+
+  private async runDSL(query: string, index: string): Promise<ApiResponse<DSLResponse>> {
+    return (await openSearchClient.transport.request({
+      method: 'GET',
+      path: index + '/_search',
+      body: query,
+    })) as ApiResponse<DSLResponse>;
```

Comment:
there's a search response interface provided `import { SearchResponse } from '@opensearch-project/opensearch/api/types';`
```suggestion
    })) as ApiResponse<SearchResponse>;
```

Reply:
Thanks for reminding me. I will change the type.